### PR TITLE
Initial commit of all QoS and feature list

### PR DESCRIPTION
**Pull request by AkhilTThomas**
This is collection of all features and QoS currently supported by cyclonedds , most of the info is gathered by searching through old issues and PRs.
@eboasson Do you think such a list makes sense? Also it would be good if you double check this list to be sure.

### CODE REVIEW

File: content/features.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+| GROUP_DATA 	| Yes 	| Yes 	| Yes 	|  	|
+| DURABILITY 	| Partial 	| - 	| - 	| only volatile and transient-local 	|
+| DURABILITY_SERVICE 	| Partial 	| - 	| - 	| service_cleanup_delay and resource limits are not supported 	|
+| PRESENTATION 	| Partial 	| - 	| - 	| only the default (INSTANCE, non-coherent, non-ordered) 	|
+| DEADLINE 	| No 	| No 	| No 	|  	|
+| LATENCY_BUDGET 	| Yes 	| Yes 	| Yes 	|  	|
+| OWNERSHIP 	| Yes 	| Yes 	| Yes 	|  	|
+| OWNERSHIP_STRENGTH 	| Yes 	| Yes 	| Yes 	|  	|
+| LIVELINESS 	| Yes 	| Yes 	| Yes 	|  	|
+| TIME_BASED_FILTER 	| Ongoing 	| No 	| No 	| [cyclonedds#1506](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1506) 	|
```
Comment (eboasson) [inline content/features.md:18]: This has now been merged. (That was the reason for not reviewing earlier, but I really should have let you know!)

It also should now work on C++ and Python (they simply forward the QoS's to the C library, which implements them).

**Review by eboasson — COMMENTED** [review]
Thanks @AkhilTThomas. Yes, I think such a list makes sense, but I also think it would be best to have it as part of the documentation, rather than as a separate file. @poetinger do you agree with that? And if you do, where would you put it?

File: content/features.md (hunk 1 above)
Comment (AkhilTThomas) [reply content/features.md:18]: I dont think all features when they're implemented in C flows into C++ and python.
Example: topic_discovery was done with https://github.com/eclipse-cyclonedds/cyclonedds/pull/615 but there is no corresponding PR in the cxx repo
also: from src/ddscxx/src/org/eclipse/cyclonedds/topic/AnyTopicDelegate.cpp
```void
AnyTopicDelegate::discover_topics(
        const dds::domain::DomainParticipant& dp,
        std::vector<dds::topic::TAnyTopic<AnyTopicDelegate> >& topics,
        uint32_t max_size)
{
    (void)dp;
    (void)topics;
    (void)max_size;
    topics.clear();
    ISOCPP_THROW_EXCEPTION(ISOCPP_UNSUPPORTED_ERROR, "Discovery of AnyTopics not implemented yet");
}

Comment (eboasson) [reply content/features.md:18]: > I dont think all features when they're implemented in C flows into C++ and python.

That's certainly true. I was simply thinking of supporting a "simple" QoS, where the only interface is the QoS setting itself. "Time based filter" is like that, so then it all falls into place rather easily.

Things like topic discovery are much more involved, they involve "real" implementation work 🙂

**Pull request closed without merging**